src/instructlab/server.py

```diff
@@ -71,8 +72,8 @@ def ensure_server(
         return (None, None)
     except ClientException:
         tried_ports = set()
-        # TODO: use default server, "spawn" doesn't work?
-        mpctx = multiprocessing.get_context(None)
+        # TODO: use fork, "spawn" fails with semaphore FileNotFound on Linux.
```

Comment:
I run on Linux and I can't repro this. I thought only the CI (Linux runner) was affected by this? Am I missing something?

Reply:
AFAIK nobody has been able to reproduce the issue locally. I have tried it multiple times on my work machine (Fedora 39). My best guess it that is is either a race condition/timing issue on GHA, a problem with GHA's runner configuration, or Ubuntu's build of Python.

Multiprocessing in spawn mode uses named semaphores (`sem_open`) to synchronize processes. Somehow the semaphore file gets removed or is not accessible from the child process (I think).

Reply:
@tiran I just saw this happen locally on a local RHEL machine @tkatarki is using.

Reply:
well to be clear, I meant the original bug just happened.

The problem I'm seeing in CI using the latest code in this PR looks like a race condition since it happened in only one of the several jobs that run the functional tests.

Reply:
and in some face-to-face discussion, a few folks said they've seen this original bug happen sometimes on Mac, as well. I had only seen it on Linux.